salesagility/SuiteCRM
# Fix #10249 https://github.com/salesagility/SuiteCRM-Core/issues/447

Fix #10249 and https://github.com/salesagility/SuiteCRM-Core/issues/447

Adds required `samesite` parameter to `SugarApplication::setCookie()`





## Description




Adds parameter `samesite` to the function `SugarApplication::setCookie()`.
Default value is set to `Strict`.

## Motivation and Context

Browsers now require `samesite` attribute on cookies, or else they'll be set to a default value `Lax` which allows a degree of third party access to the application's user login cookies.  Setting this to `Strict` by default, allows only the application domain to access application cookies, some of which are sensitive, such as user login cookies.

## How To Test This

Before this change.  Browse in the application in Chrome and Firefox.  You'll see console warnings about soon ending support for cookies that do no have the attribute `samesite`.  Checking with an extension "Cookies and Headers Analyzer" should show many cookies used by the application which have "secure" set to "false", and "samesite" unset.
After the change.  Those browser console warnings should be cleared.  Cookies used by the application should be all "secure" set to "true" and "samesite" set to "Strict".

NOTE: for those running Suite on Apache in `http` mode behind a TLS proxy, this should continue to work, because the attribute "secure" is only set to "true" when the function `isSSL()` returns true, which should be when the Suite Apache server is set to secure (SSL or TLS) mode, running on `https` port 443.

NOTE 2: There may be scenarios where setting cookie attribute "samesite" to "Strict" may break third party code, however, there must be other ways of passing information rather than allowing a third party application to have access to the application's user login cookies and have the ability to hijack user sessions and obtain sensitive organization data.

## Types of changes


### Final checklist
